packages/astro/src/types/public/internal.ts

```diff
@@ -36,21 +36,102 @@ export interface SSRLoadedRendererValue {
 	renderHydrationScript?: () => string;
 }
 
+/**
+ * It contains the information about a route
+ */
 export interface RouteData {
+	/**
+	 * The current **pattern** of the route. For example:
+	 * - `src/pages/index.astro` has a pattern of `/`
+	 * - `src/pages/blog/[...slug].astro` has a pattern of `/blog/[...slug]`
+	 * - `src/pages/site/[blog]/[...slug].astro` has a pattern of `/site/[blog]/[...slug]`
+	 */
 	route: string;
+	/**
+	 *  Source component URL 
```

Comment:
It might be worth documenting what value will be here in case of `redirect` and `fallback` routes since this is not optional and yet those types don't necessarily have a component.

Reply:
@Fryuni what would you propose then?

Reply:
Since this is for next major I think it would be better to change the name of this field to something like `subject` and then document what it means for different route types.

- For `page` routes, it is the Astro component to be rendered.
- For `endpoint` routes, it is the file declaring the method handlers.
- For `redirect` routes, it is the target mask of the redirect.
- For `fallback` routes, it is the subject of the route used as fallback.